examples/slider/js/slider-valuetext.js

```diff
@@ -0,0 +1,578 @@
+
+    let height = this.railHeight - this.thumbHeight + this.borderWidth2;
+
+    pos = this.railY + height - 1;
+    pos -= Math.round(((value - valueMin) * height) / (valueMax - valueMin));
+    this.sliderNode.setAttribute('y', pos);
+
+    // update INPUT, label and ARIA attributes
+    this.sliderValueNode.textContent = valueOutput;
+
+    // move the SVG focus ring and thumb elements
+    this.sliderFocusNode.setAttribute(
+      'y',
+      pos - (this.focusHeight - this.thumbHeight) / 2
+    );
+    this.sliderThumbNode.setAttribute('y', pos);
+
+    // Position value
+    this.sliderValueNode.setAttribute(
+      'y',
+      pos -
+        this.borderWidth +
+        this.thumbHeight -
+        (this.valueHeight - this.thumbHeight) / 2
+    );
+  }
+
+  onSliderKeydown(event) {
+    var flag = false;
+    var value = this.getValue();
+    var valueMin = this.getValueMin();
+    var valueMax = this.getValueMax();
+
+    switch (event.key) {
+      case 'ArrowLeft':
+      case 'ArrowDown':
+        this.moveSliderTo(value - this.changeValue);
+        flag = true;
+        break;
+
+      case 'ArrowRight':
+      case 'ArrowUp':
+        this.moveSliderTo(value + this.changeValue);
+        flag = true;
+        break;
+
+      case 'PageDown':
+        this.moveSliderTo(value - this.bigChangeValue);
+        flag = true;
+        break;
+
+      case 'PageUp':
+        this.moveSliderTo(value + this.bigChangeValue);
+        flag = true;
+        break;
+
+      case 'Home':
+        this.moveSliderTo(valueMin);
+        flag = true;
+        break;
+
+      case 'End':
+        this.moveSliderTo(valueMax);
+        flag = true;
+        break;
+
+      default:
+        break;
+    }
+
+    if (flag) {
+      event.preventDefault();
+      event.stopPropagation();
+    }
+  }
+
+  onSliderFocus() {
+    this.domNode.classList.add('focus');
+  }
+
+  onSliderBlur() {
+    this.domNode.classList.remove('focus');
+  }
+
+  calcValue(y) {
+    let min = this.getValueMin();
+    let max = this.getValueMax();
+    let diffY = y - this.railY;
+    return max - (diffY * (max - min)) / this.railHeight;
+  }
+
+  onRailClick(event) {
+    var value = this.calcValue(this.getSVGPoint(event).y);
+    this.moveSliderTo(value);
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    // Set focus to the clicked handle
+    this.sliderNode.focus();
+  }
+
+  onSliderPointerDown(event) {
+    this.isMoving = true;
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    // Set focus to the clicked handle
+    this.sliderNode.focus();
+  }
+
+  onPointerMove(event) {
+    if (this.isMoving) {
+      var value = this.calcValue(this.getSVGPoint(event).y);
+      this.moveSliderTo(value);
+
+      event.preventDefault();
+      event.stopPropagation();
+    }
+  }
+
+  onPointerUp() {
+    this.isMoving = false;
+  }
+}
+
+/*
+ *   Desc: Slider widget that implements ARIA Authoring Practices
+ */
+
+class SliderSeek {
```

Comment:
I'd suggest splitting each class to its own JS file, just to make them easier to find and read, but since it's already been suggested that we only have one example per page, that's probably happening anyway.